src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -58,76 +58,71 @@
-                  actionProcessingService.processActionCancel(action);
-                  successesForActionCancels++;
-                }
-              } catch (Exception e) {
-                log.error("Exception {} thrown processing action {}. Processing will be retried at next scheduled "
-                        + "distribution", e.getMessage(), action.getId());
-                log.error("Stacktrace: ", e);
-              }
-            }
-
-            try {
-              actionDistributionListManager.deleteList(actionType.getName(), true);
-            } catch (LockingException e) {
-              log.error("Failed to remove the list of actions just processed from distributed list - "
-                  + "actions distributed OK, but underlying problem may remain");
-              log.error("Stacktrace: ", e);
-            }
-          }
-
-          try {
-            actionDistributionListManager.unlockContainer();
-          } catch (LockingException e) {
-            // oh well - it will unlock soon enough
-          }
-
-          distInfo.getInstructionCounts().add(new InstructionCount(actionType.getName(),
-              DistributionInfo.Instruction.REQUEST, successesForActionRequests));
-          distInfo.getInstructionCounts().add(new InstructionCount(actionType.getName(),
-              DistributionInfo.Instruction.CANCEL_REQUEST, successesForActionCancels));
-        }
+    List<ActionType> actionTypes = actionTypeRepo.findAll();
+
+    if (!CollectionUtils.isEmpty(actionTypes)) {
+      for (ActionType actionType : actionTypes) {
+        List<InstructionCount> instructionCounts = processActionType(actionType);
+        distInfo.getInstructionCounts().addAll(instructionCounts);
       }
-    } catch (Exception e) {
-      log.error("Failed to process actions because {}", e.getMessage());
-      log.error("Stacktrace: ", e);
     }
 
     log.debug("ActionDistributor going back to sleep");
     return distInfo;
   }
 
+  private List<InstructionCount> processActionType(ActionType actionType) throws LockingException {
+    log.debug("Dealing with actionType {}", actionType.getName());
+    InstructionCount requestCount = InstructionCount.builder()
+            .actionTypeName(actionType.getName())
+            .instruction(DistributionInfo.Instruction.REQUEST)
+            .count(0)
+            .build();
+    InstructionCount cancelCount = InstructionCount.builder()
+            .actionTypeName(actionType.getName())
+            .instruction(DistributionInfo.Instruction.CANCEL_REQUEST)
+            .count(0)
+            .build();
+
+    try {
+      List<Action> actions = retrieveActions(actionType);
+
+      if (!CollectionUtils.isEmpty(actions)) {
+        processActions(actions, requestCount, cancelCount);
+      }
+    } catch (Exception e) {
+      log.error("Failed to process action type {}", actionType, e);
+      return Arrays.asList(requestCount, cancelCount);
```

Comment:
I think this return here is now redundant - if you delete it, the return after the finally will get called and the code will be clearer

Reply:
Yep you're right, done